src/pokemon.c

```diff
@@ -2791,6 +2791,8 @@ u32 GetBoxMonData3(struct BoxPokemon *boxMon, s32 field, u8 *data)
             {
                 const u8 *types = gSpeciesInfo[substruct0->species].types;
                 retVal = (boxMon->personality & 0x1) == 0 ? types[0] : types[1];
+                // To avoid this value changing upon form change/evolution, we directly set it for future cases
+                SetBoxMonData(boxMon, MON_DATA_TERA_TYPE, &retVal);
```

Comment:
I'm confused. How did this previously work if we didn't set the value?

Reply:
Because it would just use the personality each time. This provides consistent output for a single species, but as soon as the type changes (because the species changes), it changes as well.